core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -993,6 +1008,44 @@ class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSuga
       }
     }
 
+    "create a new snapshot reference when called with workspaceId" in withMinimalTestDatabase { _ =>
```

Comment:
I'm happy to add more/different tests; most of my new code should use the same code paths so I've only thrown in a minimal happy-path test just like an existing one to verify the new methods work at all.